test/unit_tests/braket/experimental/autoqasm/test_parameters.py

```diff
@@ -971,3 +971,24 @@ def parametric(phi):
 bit __bit_2__;
 __bit_2__ = measure __qubits__[0];"""
     assert parametric.to_ir() == expected
+
+
+def test_parameter_expressions_range_index():
+    """Test expressions of free parameters contained in a range index."""
+
+    @aq.main
+    def two_n_hs(n: int):
+        for _ in aq.range(2 * n):
```

Comment:
Not sure I follow

Reply:
The issue was from Printer visit_BinaryExpression trying to evaluate the precedence of "_FreeParameterExpressionIdentifier" [here](https://github.com/openqasm/openqasm/blob/main/source/openqasm/openqasm3/printer.py#L316) before visiting the node. Seems to me it would be difficult to get any custom logic to handle the `_FreeParameterExpressionIdentifier` visit since that happens after the error.